libs/azure-storage/tests/utils.py

```diff
@@ -84,3 +99,52 @@ def get_test_blobs(
         return [blob for blob in _TEST_BLOBS if blob["blob_name"].startswith(prefix)]
 
     return _TEST_BLOBS
+
+
+def get_datalake_test_blobs(
+    blob_names: Optional[Union[str, Iterable[str]]] = None, prefix: Optional[str] = None
+) -> list[dict[str, Any]]:
+    if blob_names is not None:
+        if isinstance(blob_names, str):
+            blob_names = [blob_names]
+        updated_list = []
+        for name in blob_names:
+            for blob in _TEST_DATALAKE_BLOBS:
+                if blob["blob_name"] == name:
+                    updated_list.append(
+                        {
+                            **blob,
+                            "size": len(blob["blob_content"]),
+                            "metadata": blob.get("metadata", {}),
+                        }
+                    )
+                    break
+        return updated_list
+
+    if prefix is not None:
+        return [
+            {
+                **blob,
+                "size": len(blob["blob_content"]),
+                "metadata": blob.get("metadata", {}),
+            }
+            for blob in _TEST_DATALAKE_BLOBS
+            if blob["blob_name"].startswith(prefix)
+        ]
+
+    return [
+        {
+            **blob,
+            "size": len(blob["blob_content"]),
+            "metadata": blob.get("metadata", {}),
+        }
+        for blob in _TEST_DATALAKE_BLOBS
+    ]
+
+
+def get_expected_datalake_blobs() -> list[dict[str, Any]]:
```

Comment:
Instead of exposing a new utility, let's add a new parameter `include_directories` boolean that defaults to `False` to the `get_datalake_test_blobs()` to make it a little more explicit when ADLS directories are included in the list.